src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -947,8 +948,10 @@ String getResponseToken(String responseInbox) {
     void cleanResponses(boolean cancelIfRunning) {
         ArrayList<String> toRemove = new ArrayList<>();
 
+        long cleanMillis = this.options.getRequestCleanupInterval().toMillis();
+
         responses.forEach((token, f) -> {
-            if (f.isDone() || cancelIfRunning) {
+            if (f.isDone() || f.isOlderThan(cleanMillis) || cancelIfRunning) {

```

Comment:
options.getRequestCleanupInterval() (line 951) defaults to 5 seconds. It seems to me that if we haven't gotten a response in 5 seconds, it's fair to say it's not coming back. I put this in b/c this could be a memory leak, because the future might never be done (isDone, line 954)

Reply:
Does this mean the lesser of RequestCleanupInterval or original request timeout?  Let's say I had a 30s ack timeout on a JS publish... wouldn't this expire in 5?

Reply:
I guess I could set it to something like 5 or 10 minutes. We just have to clear these eventually.

Reply:
How about setting it to the timeout used (user provided or default) + 250ms or something, and hold that value in the future itself?